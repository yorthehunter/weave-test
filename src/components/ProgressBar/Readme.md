Determinate Progress Bar using the `progress` prop, written as a percentage

```js
<ProgressBar progress={60} />
```

Indeterminate Progress Bar

```js
<ProgressBar />
```
