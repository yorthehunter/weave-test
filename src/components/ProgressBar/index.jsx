import React from 'react';
import { string, number } from 'prop-types';
import CustomTag from '../utilities/CustomTag';

const componentPrefix = 'weave-progress';


/**
 * Use a progress bar when you need to show that progress is being made after
 * an action has occurred. There is both a determinate and an indeterminate
 * style, based on whether the progress is known or unknown, respectively.
 */

const ProgressBar = (props) => {
  const {
    progress,
  } = props;

  const componentClasses = [
    componentPrefix,
  ];

  return (
    <CustomTag componentClasses={componentClasses} {...props}>
      <div
        className={progress ? `${componentPrefix}__determinate` : `${componentPrefix}__indeterminate`}
        style={progress && { width: `${progress}%` }}
      />
    </CustomTag>
  );
};

const defaultProps = {
  tag: 'div',
  progress: undefined,
};

const propTypes = {
  tag: string,
  progress: number,
};

ProgressBar.defaultProps = defaultProps;
ProgressBar.propTypes = propTypes;
export default ProgressBar;
