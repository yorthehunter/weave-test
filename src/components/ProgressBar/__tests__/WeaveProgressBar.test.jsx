import React from 'react';
import renderer from 'react-test-renderer';

import ProgressBar from '../';

describe('It renders correctly', () => {
  it('with default values', () => {
    const rendered = renderer.create(<ProgressBar />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('as a progress indicator', () => {
    const rendered = renderer.create(<ProgressBar progress={25} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
