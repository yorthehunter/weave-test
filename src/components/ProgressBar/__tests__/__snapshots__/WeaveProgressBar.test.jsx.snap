// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly as a progress indicator 1`] = `
<div
  className="weave-progress"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
>
  <div
    className="weave-progress__determinate"
    style={
      Object {
        "width": "25%",
      }
    }
  />
</div>
`;

exports[`It renders correctly with default values 1`] = `
<div
  className="weave-progress"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
>
  <div
    className="weave-progress__indeterminate"
    style={undefined}
  />
</div>
`;
